contracts/QuestFactory.sol

```diff
@@ -239,16 +243,17 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
     /// @param hash_ The hash of the message
     /// @param signature_ The signature of the hash
     function mintReceipt(string memory questId_, bytes32 hash_, bytes memory signature_) public {
-        if (quests[questId_].numberMinted + 1 > quests[questId_].totalParticipants) revert OverMaxAllowedToMint();
-        if (quests[questId_].addressMinted[msg.sender] == true) revert AddressAlreadyMinted();
-        if (!QuestContract(quests[questId_].questAddress).hasStarted()) revert QuestNotStarted();
-        if (block.timestamp < QuestContract(quests[questId_].questAddress).startTime()) revert QuestNotStarted();
-        if (block.timestamp > QuestContract(quests[questId_].questAddress).endTime()) revert QuestEnded();
+        currentQuest = quests[questId_];
+        if (currentQuest.numberMinted + 1 > currentQuest.totalParticipants) revert OverMaxAllowedToMint();
+        if (currentQuest.addressMinted[msg.sender] == true) revert AddressAlreadyMinted();
+        if (!QuestContract(currentQuest.questAddress).hasStarted()) revert QuestNotStarted();
+        if (block.timestamp < QuestContract(currentQuest.questAddress).startTime()) revert QuestNotStarted();
+        if (block.timestamp > QuestContract(currentQuest.questAddress).endTime()) revert QuestEnded();
         if (keccak256(abi.encodePacked(msg.sender, questId_)) != hash_) revert InvalidHash();
         if (recoverSigner(hash_, signature_) != claimSignerAddress) revert AddressNotSigned();
 
-        quests[questId_].addressMinted[msg.sender] = true;
-        quests[questId_].numberMinted++;
+        currentQuest.addressMinted[msg.sender] = true;
+        currentQuest.numberMinted++;
```

Comment:
I think we can save a little bit of gas on this: https://cygaar.substack.com/p/why-i-is-cheaper-than-i-in-solidity

Just need to consider how we use it (which I don't think is an issue here): https://ethereum.stackexchange.com/questions/133161/why-does-i-cost-less-gas-than-i/133164#133164